File: api-interop-layer/src/util/util-golang/data/db_test.go
package data

import (
	"os"
	"testing"
)

func TestGetDatabaseConnectionInfo(t *testing.T) {
	// Test local defaults
	os.Clearenv()
	config := GetDatabaseConnectionInfo()

	if config.User != "drupal" {
		t.Errorf("expected user drupal, got %s", config.User)
	}
	if config.Host != "database" {
		t.Errorf("expected host database, got %s", config.Host)
	}

	// Test env override
	os.Setenv("DB_HOST", "localhost")
	config = GetDatabaseConnectionInfo()
	if config.Host != "localhost" {
		t.Errorf("expected host localhost, got %s", config.Host)
	}

	// Test Production VCAP (simplified)
	os.Setenv("API_INTEROP_PRODUCTION", "true")
	vcapJSON := `{
		"aws-rds": [{
			"credentials": {
				"username": "prod_user",
				"password": "prod_password",
				"name": "prod_db",
				"host": "prod_host",
				"port": 5432
			}
		}]
	}`
	os.Setenv("VCAP_SERVICES", vcapJSON)

	config = GetDatabaseConnectionInfo()
	if config.User != "prod_user" {
		t.Errorf("expected user prod_user, got %s", config.User)
	}
	if config.SSL != true {
		t.Errorf("expected SSL true in prod")
	}
}

File: api-interop-layer/data/db.test.js
import { expect } from "chai";
import { getDatabaseConnectionInfo } from "./db.js";

describe("database utility", () => {
  it("defaults to hardcoded connection details", () => {
    delete process.env.DB_USERNAME;
    delete process.env.DB_PASSWORD;
    delete process.env.DB_NAME;
    delete process.env.DB_HOST;
    delete process.env.DB_PORT;
    delete process.env.API_DB_MAX_CONNECTIONS;

    const actual = getDatabaseConnectionInfo();

    expect(actual).to.eql({
      user: "drupal",
      password: "drupal",
      database: "weathergov",
      host: "database",
      port: 3306,
      min: 22,
      max: 45,
    });
  });

  it("uses environment variables if set and not in production", () => {
    process.env.DB_USERNAME = "Link";
    process.env.DB_PASSWORD = "It's a secret to everybody.";
    process.env.DB_NAME = "Triforce";
    process.env.DB_HOST = "Hyrule";
    process.env.DB_PORT = "Zora's Domain";
    process.env.API_DB_MAX_CONNECTIONS = "100";

    const actual = getDatabaseConnectionInfo();

    expect(actual).to.eql({
      user: "Link",
      password: "It's a secret to everybody.",
      database: "Triforce",
      host: "Hyrule",
      port: "Zora's Domain",
      min: 50,
      max: 100,
    });
  });

  it("uses VCAP data if in production", () => {
    delete process.env.API_DB_MAX_CONNECTIONS;

    process.env.API_INTEROP_PRODUCTION = "true";
    process.env.VCAP_SERVICES = JSON.stringify({
      "aws-rds": [
        {
          credentials: {
            username: "cloudy bunch",
            password: "carlo in particular",
            name: "postgres",
            host: "somewhere in an AWS datacenter",
            port: "any, in a storm",
            min: 20,
            max: 40,
          },
        },
      ],
    });

    const actual = getDatabaseConnectionInfo();

    expect(actual).to.eql({
      user: "cloudy bunch",
      password: "carlo in particular",
      database: "postgres",
      host: "somewhere in an AWS datacenter",
      port: "any, in a storm",
      ssl: true,
      min: 97,
      max: 195,
    });

    delete process.env.API_INTEROP_PRODUCTION;
  });
});
